### PR TITLE
DM-49265: Test and fix for incorrect connections in DRP-v2.

### PR DESCRIPTION
**Pull request by TallJimbo**

### CODE REVIEW

**Review by enourbakhsh — APPROVED** [review]
Looks good overall! Just pointed out a few small things (mostly about logging) to make sure nothing is missed.

File: python/lsst/drp/pipe/tests/correspondence.py [hunk 1]
```diff
@@ -124,15 +128,24 @@ def sorted(self, new: PipelineGraph, old: PipelineGraph) -> Correspondence:
...
         return result
 
-    def check(self, new: PipelineGraph, old: PipelineGraph, new_name: str, old_name: str) -> list[str]:
+    def check(
+        self,
+        new: PipelineGraph,
+        old: PipelineGraph,
+        new_name: str,
+        old_name: str,
+        ignore_edges: Set[tuple[str, str]] = frozenset(),
```
[the remark this answers is not in the document]
Comment (TallJimbo) [reply python/lsst/drp/pipe/tests/correspondence.py:147]: I wondered about that, but the logs coming from this test are already so verbose I think we wouldn't actually be able to use it to spot anything.

File: python/lsst/drp/pipe/tests/correspondence.py [hunk 2]
```diff
@@ -154,7 +172,10 @@ def check(self, new: PipelineGraph, old: PipelineGraph, new_name: str, old_name:
             problems.
         """
         messages: list[str] = []
-        tasks_old_to_new = {old_label: new_label for new_label, old_label in self.tasks_new_to_old.items()}
+        tasks_old_to_new = {
+            old_label: new_label
+            for new_label, old_label in self.tasks_new_to_old.items()
+        }
```
[the remark this answers is not in the document]
Comment (TallJimbo) [reply python/lsst/drp/pipe/tests/correspondence.py:178]: The check right below this line guards against this: if there is a duplicate in the values, the length of the reversed mapping will differ from the original.

File: python/lsst/drp/pipe/tests/correspondence.py [hunk 3]
```diff
@@ -206,26 +232,69 @@ def check(self, new: PipelineGraph, old: PipelineGraph, new_name: str, old_name:
...
+                    if mapped_old_name := self.dataset_types_new_to_old.get(new_name):
+                        if not (old_edge := old_edge_map.get(connection_name)):
+                            # Some analysis_tools tasks change their connection
+                            # name when configured differently, but use the
+                            # dataset type name as the connection.
+                            old_connection_name = mapped_old_name
+                            if not (old_edge := old_edge_map.get(old_connection_name)):
+                                # This is some other kind of dynamic connection
+                                # that we just can't check.
+                                continue
```
[the remark this answers is not in the document]
Comment (TallJimbo) [reply python/lsst/drp/pipe/tests/correspondence.py:291]: It's just an edge case we can't test.  I think it's sufficiently rare that I'm not worried about it.
